File: products-list/src/main/java/com/github/zeroberto/productsstore/model/Product.java
package com.github.zeroberto.productsstore.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import static lombok.AccessLevel.PRIVATE;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor(access = PRIVATE)
public class Product {

  private String id;
  private int priceInCents;
  private String title;
  private String description;

}

File: products-list-endpoint/model/product.go
package model

// Product is a model structure that represents a store product
type Product struct {
	ID           string    `json:"id"`
	PriceInCents int32     `json:"price_in_cents"`
	Title        string    `json:"title"`
	Description  string    `json:"description"`
	Discount     *Discount `json:"discount"`
}
